filebeat/inputsource/tcp/client.go

```diff
@@ -73,7 +73,9 @@ func (c *client) handle() error {
 	buf := bufio.NewReader(r)
 	scanner := bufio.NewScanner(buf)
 	scanner.Split(c.splitFunc)
-
+	//16 is ratio of MaxScanTokenSize/startBufSize
+	buffer := make([]byte, c.maxMessageSize/16)
```

Comment:
Why 16?

Reply:
Bufio scan uses default buffer size as 4096 and `MaxScanTokenSize` as 64 * 1024. I thought, when we set maxMessageSize to  `MaxScanTokenSize`, I should keep the ratio between maxScantoken size and buffer size. That is why I divided maxMessageSize to 16.

Reply:
@kvch I have a question for you. When I run in my local, I m not having any issue on unit tests. I have tried in my macos and Linux env. Both were fine. But for some reason, it is failing in travis and jenkins. Am I missing some configuration or config file? I m using `go test -race -cover ./...` under filebeat folder to run the unit testes.

Reply:
I could able to address the problem when I run with `make testsuite` :)